modules/node/handler/nodes_handler.e

```diff
@@ -49,10 +52,68 @@ feature -- HTTP Methods
 			create {GENERIC_VIEW_CMS_RESPONSE} l_page.make (req, res, api)
 			l_page.add_variable (node_api.nodes, "nodes")
 
+			create pager.make (api, node_api)
+			number_of_pages := (node_api.nodes_count // pager.limit) + 1
+
+				-- Size:limit
+			if
+				attached {WSF_STRING} req.query_parameter ("size") as l_size and then
+				l_size.is_integer
+			then
+				pager.set_limit (l_size.integer_value.to_natural_32)
+			end
+
+
+
+				--Page:offset
+			if
+				attached {WSF_STRING} req.query_parameter ("page") as ll_page and then
+				ll_page.is_integer
+			then
+				current_page := ll_page.integer_value
+				if current_page > 1 then
+					pager.set_offset (((current_page-1)*(pager.limit.to_integer_32)).to_natural_32)
+				end
+			else
+				current_page := 1
+			end
+
+
 
 				-- NOTE: for development purposes we have the following hardcode output.
 			create s.make_from_string ("<p>Nodes:</p>")
-			if attached node_api.nodes as lst then
+
+			s.append ("<p>Current Page:" + current_page.out +  " of " + number_of_pages.out + " pages</p>" )
```

Comment:
not critical but
s.append ("<p>Current Page:" + current_page.out +  " of " + number_of_pages.out + " pages</p>" )
could be written

s.append ("<p>Current Page:")
s.append_natural (current_page)
s.append (" of ")
s.append_natural (number_of_pages)
s.append (" pages</p>" )

this avoids creating many intermediate temp string object.